extensions/azurecore/src/azureResource/providers/database/databaseService.ts

```diff
@@ -19,18 +20,27 @@ export class AzureResourceDatabaseService implements IAzureResourceService<azure
 		const resourceClient = new ResourceGraphClient(credential, { baseUri: account.properties.providerSettings.settings.armResource.endpoint });
 
 		// Query servers and databases in parallel (start both promises before waiting on the 1st)
-		let serverQueryPromise = queryGraphResources<GraphData>(resourceClient, subscriptions, serversQuery);
+		let synapseQueryPromise = queryGraphResources<GraphData>(resourceClient, subscriptions, synapseWorkspacesQuery);
+		let serverQueryPromise = queryGraphResources<GraphData>(resourceClient, subscriptions, sqlServersQuery);
 		let dbQueryPromise = queryGraphResources<GraphData>(resourceClient, subscriptions, `where type == "${azureResource.AzureResourceType.sqlDatabase}"`);
-		let servers: DbServerGraphData[] = await serverQueryPromise as DbServerGraphData[];
+		let server1: (DbServerGraphData | DbSynapseGraphData)[] = await serverQueryPromise as DbServerGraphData[];
+		let server2: (DbServerGraphData | DbSynapseGraphData)[] = await synapseQueryPromise as DbSynapseGraphData[];
 		let dbByGraph: DatabaseGraphData[] = await dbQueryPromise as DatabaseGraphData[];
+		let servers: (DbServerGraphData | DbSynapseGraphData)[] = server1.concat(server2);
 
 		// Group servers by resource group, then merge DB results with servers so we
 		// can get the login name and server fully qualified name to use for connections
-		let rgMap = new Map<string, DbServerGraphData[]>();
+		let rgMap = new Map<string, (DbServerGraphData | DbSynapseGraphData)[]>();
 		servers.forEach(s => {
-			let serversForRg = rgMap.get(s.resourceGroup) || [];
-			serversForRg.push(s);
-			rgMap.set(s.resourceGroup, serversForRg);
+			if ((s as any).properties.connectivityEndpoints) {
```

Comment:
`(s as DbSynapseGraphData).properties.connectivityEndpoints` 

Using any loses you the autocompletion and type checking. 

Same for all the other places you're doing similar checks.